io/plugins/eu.esdihumboldt.hale.io.shp/src/eu/esdihumboldt/hale/io/shp/writer/ShapefileInstanceWriter.java

```diff
@@ -724,4 +730,35 @@ private List<String> writeToFile(
 		return filesWritten;
 	}
 
+	/**
+	 * Create the CPG file starting from the Shapefile
+	 * 
+	 * @param cpgFilePath Path of the file to be written with just one line of
+	 *            the encoding
+	 * @throws IOException exception in any.
+	 */
+	public void writeCodePageFile(String cpgFilePath) throws IOException {
+		File cpgFile = new File(cpgFilePath);
+		FileWriter fileWriter = new FileWriter(cpgFile);
+
+		ShapefileDataStore store = new ShapefileDataStore(
+				new File(cpgFilePath.replace(ShapefileConstants.CPG_EXTENSION,
+						ShapefileConstants.SHP_EXTENSION)).toURL());
+
+		try {
+			fileWriter.write(store.getCharset() != null ? store.getCharset().toString()
```

Comment:
Looking at the code for [ShapefileDataStore](https://github.com/geotools/geotools/blob/5e3616c984b432573e2e58ad51b133f2580ddb0e/modules/plugin/shapefile/src/main/java/org/geotools/data/shapefile/ShapefileDataStore.java#L98C32-L98C32) this would then always be the default value from there (ISO 8859-1).

I preferred the solution from before, using the writer setting - this would then have opened the possibility for users to specify the desired encoding. As mentioned before, that would have required making sure that this charset is used when writing the Shapefile, which should be easily possible where the Shapefile data store for writing the respective file is obtained.

With how it is done here now, it is very intransparent which charset is used and why, and we solely rely on that the default is used here and also where the Shapefile is actually written.